packages/gatsby/cache-dir/component-renderer.js

```diff
@@ -91,10 +92,12 @@ class ComponentRenderer extends React.Component {
   }
 
   render() {
+    const pluginResponses = apiRunner(`replacePageComponentRenderer`, { props: this.props })
+    const replacementPage = pluginResponses[0]
     // If page.
     if (this.props.page) {
```

Comment:
Any reason not to just put `replacePageComponentRenderer` inside the page check?

Reply:
That makes sense for our particular use case. We don't have any plans to transition layouts. However, I didn't want to make that assumptions for others. This seemed more general at the cost of clients (such as myself) having to check if it was a layout in the runner implementation.

Reply:
Now that I've said that, it shouldn't really be named  replace**Page** then, huh?

Reply:
You cool with both `replacePageComponentRenderer` and `replaceLayoutComponentRenderer`?

Reply:
Haha yeah that's what tripped me up — `replaceComponentRenderer` would do nicely